models/superrare/superrare_ethereum_events.sql

```diff
@@ -0,0 +1,372 @@
+        select date_trunc('week',block_time) as week
+                , avg(eth_per_rare) as average_price_that_day_eth_per_rare
+        from (select * from all_rare_eth_dex_trades where eth_per_rare < 0.001) a
+        group by 1 
+        ) b 
+    on a.week = b.week
+)
+
+-- Mint details needed to determine whether a transaction was a primary or secondary sale. 
+-- Sometimes primary sales are not sold directly from the artist's address (e.g. sold from Gallery)
+, mint_address_details_per_token_id as (
+    select  contract_address
+            , `tokenId`
+            , a.`to` as mint_sent_to
+            , b.`from` as mint_created_by
+    from 
+    (
+    select contract_address
+            , `tokenId`
+            , evt_tx_hash
+            , `to` 
+    -- from erc721_ethereum.evt_Transfer
+    from {{ source('erc721_ethereum','evt_transfer') }}
+    where contract_address in (select distinct contract_address from all_superrare_sales)
+    and `tokenId` in (select distinct `_tokenId` from all_superrare_sales)
+    and `from` = '0x0000000000000000000000000000000000000000'
+    ) a 
+    inner join 
+    ( select `from` 
+            , hash 
+    -- from ethereum.transactions
+    from {{ source('ethereum','transactions') }}
+    ) b 
+    on a.evt_tx_hash = b.hash
+)
+, mint_address_per_token_20 as (
+    select contract_address
+            , `value` as `tokenId`
+            , `to` as mint_address 
+    -- from erc20_ethereum.evt_Transfer
+    from {{ source('erc20_ethereum','evt_transfer') }}
+    where contract_address in (select distinct contract_address from all_superrare_sales)
+    and `value` in (select distinct `_tokenId` from all_superrare_sales)
+    and `from` = '0x0000000000000000000000000000000000000000'
+)
+, token_sold_from_auction as (
+    select contract_address
+            , `_tokenId`
+    from all_superrare_sales
+    where `_seller` = lower('0x8c9f364bf7a56ed058fc63ef81c6cf09c833e656')
+)
+, transfers_for_tokens_sold_from_auction as (
+        select contract_address
+            , `tokenId`
+            , evt_tx_hash
+            , `to` 
+            , `from`
+            , lag(`from`) OVER (PARTITION BY contract_address, `tokenId` ORDER BY evt_block_time asc) as previous_owner
+            , case  when `from` = lower('0x8c9f364bf7a56ed058fc63ef81c6cf09c833e656')
+                        then 'From Auction House'
+                    when `to` = lower('0x8c9f364bf7a56ed058fc63ef81c6cf09c833e656')
+                        then 'To Auction House'
+                else '' end as auction_house_flag
+            , ROW_NUMBER() OVER (PARTITION BY contract_address, `tokenId` ORDER BY evt_block_time DESC) AS transaction_rank
+    -- from erc721_ethereum.evt_Transfer
+    from {{ source('erc721_ethereum','evt_transfer') }}
+    where contract_address in (select distinct contract_address from token_sold_from_auction)
+    and `tokenId` in (select distinct `_tokenId` from token_sold_from_auction)
+    order by 1, 2, 8
```

Comment:
{% if is_incremental() %} -- this filter will only be applied on an incremental run
    and evt_block_time >= date_trunc("day", now() - interval '1 week')
    {% endif %}